plugins/audit_logs/scripts/report.py

```diff
@@ -0,0 +1,87 @@
+"""
+This script is for creating download reports for a given folder.
+
+    girder shell --plugins=audit_logs report.py -- --folder=57557fac8d777f68be8f3f49 --start-date=2018-09-10T13:55:34.847Z --end-date=2018-09-13T13:55:34.847Z report.csv
+"""
+import click
+import datetime
+import dateutil.parser
+
+from bson.objectid import ObjectId
+
+from girder.models.item import Item
+from girder.models.folder import Folder
+from girder.constants import AccessType
+from girder.plugins.audit_logs import Record
+
+
+def index_folder(folderId):
+    if Folder().load(folderId, force=True) is None:
+        raise ValueError('folderId={} was not a valid folder'.format(folderId))
+    items = Item().find({'folderId': ObjectId(folderId)})
+    subfolders = Folder().find({'parentId': ObjectId(folderId)})
+
+    files = []
+    for item in items:
+        for file in Item().childFiles(item, fields={'_id': True}):
+            fileId = file['_id']
+            files.append(fileId)
+
+    for folder in subfolders:
+        files += index_folder(folder['_id'])
+
+    return files
+
+
+def get_file_download_records(files, start=None, end=None):
+    query = {
+        'type': 'file.download',
+        'details.fileId': {
+            '$in': files,
+        },
+        'details.startByte': 0,
+        'when': {
+            '$gte': start,
+            '$lt': end,
+        }
+    }
+    if (start is not None) or (end is not None):
+        whenClause = { 'when': {} }
+        if start is not None:
+            whenClause['when']['$gte'] = dateutil.parser.parse(start)
+        if end is not None:
+            whenClause['when']['$lt'] = dateutil.parser.parse(end)
+        query.update(whenClause)
+    return Record().find(query)
+
+
+def csv_report_generator(files, records):
+    """
+    :param files: a list of objectId strings\e
+    :param records: a mongo cursor with all download records
+    :return: an output generator
+    """
+    yield 'file_id,ip,timestamp'
+    for r in records:
+        yield '{id},{address},{timestamp}'.format(
+            id=r['details']['fileId'],
+            address=r['ip'],
+            timestamp=str(r['when']),
```

Comment:
Let's change this to `timestamp=r['when'].isoformat()` to be strict in what we emit.